src/KilnUniV3.sol

```diff
@@ -0,0 +1,123 @@
+// https://docs.uniswap.org/protocol/guides/swaps/multihop-swaps#input-parameters
+struct ExactInputParams {
+    bytes   path;
+    address recipient;
+    uint256 deadline;
+    uint256 amountIn;
+    uint256 amountOutMinimum;
+}
+
+contract KilnUniV3 is KilnBase, TwapProduct {
+    uint256 public scope; // [Seconds]  Time period for TWAP calculations
+    uint256 public yen;   // [WAD]      Relative multiplier of the TWAP's price to insist on
+    bytes   public path;  //            ABI-encoded UniV3 compatible path
+
+    address public immutable uniV3Router;
+    address public immutable receiver;
+
+    event File(bytes32 indexed what, bytes data);
+
+    // @notice initialize a Uniswap V3 routing path contract
+    // @dev TWAP-relative trading is enabled by default. With the initial values, fire will 
+    //      perform the trade only when the amount of tokens received is equal or better than
+    //      the 1 hour average price.
+    //      file("yen", 0) to disable TWAP trading and accept any output amount.
+    // @param _sell the contract address of the token that will be sold
+    // @param _buy  the contract address of the token that will be purchased
+    // @param _uniV3Router the address of the current Uniswap V3 swap router
+    // @param _receiver the address of the account which will receive the funds to be bought
+    constructor(
+        address _sell,
+        address _buy,
+        address _uniV3Router,
+        address _receiver
+    )
+        KilnBase(_sell, _buy)
+        TwapProduct(SwapRouterLike(_uniV3Router).factory())
+    {
+        uniV3Router = _uniV3Router;
+        receiver    = _receiver;
+
+        scope = 1 hours;
+        yen = WAD;
+    }
+
+    uint256 constant WAD = 10 ** 18;
+
+    /**
+        @dev Auth'ed function to update path value
+        @param what   Tag of value to update
+        @param data   Value to update
+    */
+    function file(bytes32 what, bytes calldata data) external auth {
+        if (what == "path") path = data;
+        else revert("KilnUniV3/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    /**
+        @dev Auth'ed function to update yen, scope, or base contract derived values
+        @param what   Tag of value to update
+        @param data   Value to update
+    */
+    function file(bytes32 what, uint256 data) public override auth {
+        if      (what == "yen") yen = data;
+        else if (what == "scope") {
+            require(data <= type(uint32).max, "KilnUniV3/scope-overflow");
+            scope = data;
+        } else {
+            super.file(what, data);
+            return;
+        }
+        emit File(what, data);
+    }
+
+    function _swap(uint256 amount) internal override returns (uint256 swapped) {
+        require(GemLike(sell).approve(uniV3Router, amount));
+
+        bytes memory _path = path;
+        uint256      _yen  = yen;
```

Comment:
```suggestion
        bytes    memory _path = path;
        uint256               _yen  = yen;
```

Reply:
I think this is now handled in f27eb9172.